kiali/kiali
# Hide mesh page for non istio-system users

### Describe the change

- Hide mesh page for non istio-system users
- Don't perform istio/status call if Kiali doesn't have access to the control plane
- Remove the link to the Mesh page from the About window

### Steps to test the PR

It can be tested with token strategy. 
Use a kiali token: 
`
k create token kiali -n istio-system`

The user should see the mesh page, and the Mesh page link in the Kiali about window. 

Create a user with restricted permissions: 
`kubectl create serviceaccount test`

Create role: 
`kubectl apply -f https://raw.githubusercontent.com/josunect/yamls/main/roleBookinfo.yaml`
Create roleBinding: 
`kubectl apply -f https://raw.githubusercontent.com/josunect/yamls/main/roleBindingBookinfo.yaml`

Get a token: 
`kubectl create token test`

Logging, and the user should not see the mesh page: 

![image](https://github.com/user-attachments/assets/4b2c58b5-f04b-4054-a84a-b91c9d6d78ce)

No link to Mesh in the about window: 

![image](https://github.com/user-attachments/assets/ef9146a8-c9f7-46e5-a037-f6b918e29009)


### Automation testing

 e2e tests: Added a new "authorization" feature file to test features with a user (**bookinfouser**) that has access to:

- east cluster
- bookinfo namespace (No control plane access)

This is meant to work for the multi primary scripts that set up this user and the permissions with openId.

Some initial tests: 

- The user doesn't see the Mesh menu
- The user doesn't see the istio-system card
- The user see the bookinfo card
- The user doesn't see the Mesh link in the about window

In the mesh page, a couple of tests were added: 

- The user see the Mesh menu
- The user see the Mesh link in the about window

### Issue reference

Fixes #7527
